ostd/src/arch/x86/cpu/mod.rs

```diff
@@ -45,103 +43,105 @@ pub struct CpuExceptionInfo {
     pub page_fault_addr: usize,
 }
 
+struct GeneralRegsWrapper(RawGeneralRegs);
```

Comment:
This struct is also for Intel TDX only. Wrap the type and its implementation block in a module and use a single `cfg` to control the compilation of the module.

In addition, the TDX-specific imports at the top of this file can be moved into this newly-added module and the TDX-specific code block in `UserContext::execute`.